libpod/container_log_linux.go

```diff
@@ -83,10 +83,28 @@ func (c *Container) readFromJournal(ctx context.Context, options *logs.LogOption
 	if err := journal.AddDisjunction(); err != nil {
 		return fmt.Errorf("adding filter disjunction to journald logger: %w", err)
 	}
-	match = sdjournal.Match{Field: "CONTAINER_ID_FULL", Value: c.ID()}
-	if err := journal.AddMatch(match.String()); err != nil {
-		return fmt.Errorf("adding filter to journald logger: %v: %w", match, err)
+
+	if isPassthrough {
+		// Match based on systemd unit which is the container is cgroup
+		// so we get the exact logs for a single container even in the
+		// play kube case where a single unit starts more than one container.
+		unitTypeName := "_SYSTEMD_UNIT"
+		if rootless.IsRootless() {
+			unitTypeName = "_SYSTEMD_USER_UNIT"
+		}
+		// FIXME: It looks like it is hardcoded to libpod-ID.scope but I am not
+		// sure were we set this with all the different cgroup modes???
+		match = sdjournal.Match{Field: unitTypeName, Value: "libpod-" + c.ID() + ".scope"}
+		if err := journal.AddMatch(match.String()); err != nil {
+			return fmt.Errorf("adding filter to journald logger: %v: %w", match, err)
+		}
```

Comment:
@giuseppe PTAL
Not sure if it is safe to assume libpod-<ID>.scope as cgroup name? I couldn't figure out where this is set in podman.

Reply:
with passthrough we usually run from a systemd service (and likely reusing the cgroup with `--cgroups=split`).  Isn't in this case the service name always different and we cannot guess that?

Reply:
Well I do not know how system and cgroup works but what I can tell you that this works for both the podman-kube@ unit and the a unit created with `podman generated systemd --new` with added passthrough driver. Both of them do not use --cgroups=split at the moment.

If we have a function in podman where I could get this path it would help. I found c.cGroupPath() but this only works on running containers. However it seems like I could just copy the switch case logic from there?

Reply:
I just checked it with cgroup=split the name is set to the real unit name and we know that, it is stored as `PODMAN_SYSTEMD_UNIT` label so I can/will make it work.